kernel/thread.c

```diff
@@ -1242,7 +1247,14 @@ size_t myst_kill_thread_group()
         for (t = tail; t != NULL; t = t->group_prev)
         {
             if (t != process && t != thread && t->status != MYST_ZOMBIE)
+            {
```

Comment:
may wont to delete the printf!
In my PR I did another WAKE here as there is a race condition between the signal on the killing and the waiting. The code may be in a transitional state where it has not quite made it to the wait and so the previous wake may not be enough. Adding it here also gets that other race.

Reply:
Will add this to the following loop. Thanks.